src/Expect.elm

```diff
@@ -345,20 +516,58 @@ onFail str expectation =
             Test.Expectation.Fail given str
 
 
-reportFailure : String -> String -> String -> String
+reportFailure : String -> a -> b -> String
 reportFailure comparison expected actual =
-    [ actual
+    [ toString actual
     , "╷"
     , "│ " ++ comparison
     , "╵"
-    , expected
+    , toString expected
```

Comment:
Hm, this would change the behavior of existing tests; `toString` on a string wraps it in quotes, so `toString "foo" == "foo"` is `False` but `toString "foo" == "\"foo\""` is `True`.

I think we should leave `reportFailure` as it was, and have callers do the `toString` when appropriate.